src/systems/crdt/index.ts

```diff
@@ -1,132 +1,170 @@
+   */
+  function parseChunkMessage(transportType: string) {
+    /**
+     * Receives a chunk of binary messages and stores all the valid
+     * Component Operation Messages at messages queue
+     * @param chunkMessage A chunk of binary messages
+     */
+    return function parseChunkMessage(chunkMessage: MessageEvent<Uint8Array>) {
+      if (!chunkMessage.data?.length) return
+      const buffer = createByteBuffer({
+        reading: { buffer: chunkMessage.data, currentOffset: 0 }
       })
+
+      while (WireMessage.validate(buffer)) {
+        const offset = buffer.currentReadOffset()
+        const message = Message.read(buffer)!
+
+        const { entity, componentId, data, timestamp } = message
+        receivedMessages.push({
+          entity,
+          componentId,
+          data,
+          timestamp,
+          transportType,
+          messageBuffer: buffer
+            .buffer()
+            .subarray(offset, buffer.currentReadOffset())
+        })
+      }
     }
   }
 
-  function getMessages() {
-    const messagesToProcess = Array.from(messages)
-    messages.length = 0
+  /**
+   * Return and clear the messaes queue
+   * @returns messages recieved by the transport to process on the next tick
+   */
+  function getMessages<T = unknown>(value: T[]) {
+    const messagesToProcess = Array.from(value)
+    value.length = 0
     return messagesToProcess
   }
 
   /**
-   * This messages will be processed on every tick.
+   * This fn will be called on every tick.
+   * Process all the messages queue received by the transport
    */
-  function processMessages() {
-    const buffer = createByteBuffer()
-    const messagesToProcess = getMessages()
-
-    for (const message of messagesToProcess) {
-      const [entity, componentId] = CrdtUtils.parseKey(message.key)
-      const msg = crdtClient.processMessage(message)
-      const component = engine.getComponent(componentId)
-      // CRDT outdated message. Resend this message through the wire
-      // TODO: perf transactor
-      if (msg !== message) {
-        PutComponentOperation.write(entity, msg.timestamp, component, buffer)
-      } else {
-        const bb = createByteBuffer({
-          reading: {
-            buffer: message.data,
-            currentOffset: 0
-          }
-        })
-
-        component.upsertFromBinary(entity, bb)
-        component.clearDirty()
+  function receiveMessages() {
+    const messagesToProcess = getMessages(receivedMessages)
+    for (const transport of transports) {
+      const buffer = createByteBuffer()
+      for (const message of messagesToProcess) {
+        const { data, timestamp, componentId, entity } = message
+        const crdtMessage: CrdtMessage<Uint8Array> = {
+          key: CrdtUtils.getKey(entity, componentId),
+          data: data,
+          timestamp: timestamp
+        }
+        const component = engine.getComponent(componentId)
+        const currentMessage = crdtClient.processMessage(crdtMessage)
```

Comment:
To have a better DX, may this be a boolean signaling a change in state?
```suggestion
        const didChangeState = crdtClient.processMessage(crdtMessage)
```